service/src/main/kotlin/fi/espoo/evaka/emailclient/EmailConfig.kt

```diff
@@ -8,13 +8,25 @@ import com.amazonaws.services.simpleemail.AmazonSimpleEmailService
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.core.env.Environment
+import org.springframework.core.env.getRequiredProperty
 
 @Configuration
 class EmailConfig {
     @Bean
     fun emailClient(client: AmazonSimpleEmailService, env: Environment): IEmailClient =
         when (env.getProperty("application.email.enabled", Boolean::class.java, false)) {
-            true -> EmailClient(client = client)
+            true -> EmailClient(
+                client = client,
+                whitelist = env.getRequiredProperty<List<String>>("application.email.whitelist").map { pattern ->
+                    // Assumes the patterns are of form "*", "*@test.com", etc.
+                    pattern
+                        .split("*")
+                        .joinToString(".*", "^", "$") {
+                            if (it.isNotBlank()) Regex.escape(it) else it
+                        }
+                        .let { Regex(it) }
+                }
```

Comment:
Looking at this, I feel this is quite brittle and IMO it would be better to just move the regex-reformatting into Terraform (i.e. the env vars should be expected to be valid regex as-is). In Terraform it wouldn't bee to difficult either and making assumptions about the variable format would make more sense closer to where the variables are actually defined:

```hcl
APPLICATION_EMAIL_WHITELIST = join(",", [for r in local.allowed_recipients: format("^%s$", replace(r, "*", ".*"))])
```

Reply:
Sure, that does make sense.